woocommerce/google-listings-and-ads
# Enable labeling of Ads campaigns

### Changes proposed in this Pull Request:



Part of pcTzPl-2nS-p2

We would like to track where campaigns are created from by labeling them with their source of creation. For example, use `wc-gla` for the web version, and for the mobile versions, something like `wc-gla-android` or `wc-gla-ios`.

This PR allows the creation of a new campaign with a specific label. Note that I haven't updated the frontend to send the label query parameter yet, as I would first like to define the label names we'll be using.


### Screenshots:




### Detailed test instructions:


 1. Make the following call `POST wp-json/wc/gla/ads/campaigns` with this body: 
```
{
    "amount": 0.01,
    "targeted_locations": ["ES"],
    "label": "wc-gla"
}
```
3. The campaign should be created successfully.
4. Go to https://ads.google.com/
5. Filter your campaign by label:

![image](https://github.com/user-attachments/assets/09433042-a0fd-4146-ad9e-6d07c5808f37)

You should see the campaign that you just created.
6. Alternatively, you can customize the campaigns table to display the labels:

![image](https://github.com/user-attachments/assets/0eadfd8f-66fc-4c68-b1c8-62061ebb6318)

![image](https://github.com/user-attachments/assets/b8814f46-efa8-4318-bcbe-f55a0b145675)

![image](https://github.com/user-attachments/assets/1db47049-364e-485e-aa91-879c17a8196f)



### Additional details:


### Changelog entry

> Add - Enable labeling of Ads campaigns
